src/lib.rs

```diff
@@ -2729,6 +2731,68 @@ pub fn is_maximal_matching(
     })
 }
 
+/// Find the edges in the minimum spanning tree or forest of a graph
+/// using Kruskal's algorithm.
+///
+/// :param PyGraph graph: Undirected weighted graph with comparable weights
+///
+/// :returns: The :math:`N - |c|` edges of the Minimum Spanning Tree (or Forest, if :math:`|c| > 1`)
+///     where :math:`N` is the number of nodes and :math:`|c|` is the number of connected components of the graph
+/// :rtype: WeightedEdgeList
+#[pyfunction]
+#[text_signature = "(graph,)"]
+pub fn minimum_spanning_edges(
+    py: Python,
+    graph: &graph::PyGraph,
+) -> WeightedEdgeList {
+    let mut subgraphs = UnionFind::<usize>::new(graph.graph.node_bound());
+    let mut edge_list: Vec<EdgeReference<PyObject>> =
+        graph.graph.edge_references().collect();
+
+    edge_list.sort_by(|a, b| {
+        let weight_a = a.weight().as_ref(py);
+        let weight_b = b.weight().as_ref(py);
+        weight_a.compare(weight_b).unwrap()
```

Comment:
If you change the return type to `PyResult<WeightedEdgeList>` can't you just do:

```suggestion
        weight_a.compare(weight_b)?
```

here? We don't want to `unwrap()` a potential error, especially from python. We need to preserve those Python errors as they're exceptions which need to be returned to the caller. Especially here, where you're relying on python to do the object comparison the user needs to know that they're using objects which couldn't be compared.

Reply:
We cannot use `weight_a.compare(weight_b)?` because the call is in the closure, and unfortunately the closure can't return `Result<cmp::Ordering>`, it must return `cmp::Ordering`. I will try find an alternative